### PR TITLE
Use Dataset Metadata for Datetime Handling

### PR DESCRIPTION
**Pull request by WillKoehrsen**
- Uses the `dataset_metadata` provided by the server to determine which columns should be converted to datetimes

### CODE REVIEW

File: gridstatusio/gs_client.py [hunk 1]
```diff
@@ -375,50 +381,14 @@ def get_dataset(
...
-                            columns={col: col.replace("_utc", "") + "_local"},
-                        )
+        all_columns = dataset_metadata.get("all_columns", [])
 
-                except ValueError:
-                    pass
+        # convert to datetime any columns that are datetimes
+        for col in all_columns:
+            if col["is_datetime"]:
+                df[col] = pd.to_datetime(df[col], utc=True)
```
Comment (kmax12) [inline gridstatusio/gs_client.py:391]: sorry, if im jumping in before this is finished.

we still need to do the `_local` handling with timezone conversion

Comment (WillKoehrsen) [reply gridstatusio/gs_client.py:391]: Updated.

**Comment by WillKoehrsen** [comment]
There's a situation not handled so far: when a dataset does not have `interval_start_utc`/`interval_end_utc` columns and we do resampling, those columns are added to the dataframe. Since these columns aren't in the `dataset_metadata`, they won't get converted to datetimes with the current approach. 

Should we maintain a list of columns that will always be datetimes? I think that list would just be `["interval_start_utc", "interval_end_utc"]` now.

**Comment by kmax12** [comment]
interesting, I hadn't considered that situation. I agree with your approach.

**Pull request merged**